rubin_jupyter_utils/helpers/token.py

```diff
@@ -4,27 +4,49 @@
 
 import os
 import requests
-
+from pathlib import Path
 from .log import make_logger
 
 
 def get_access_token(tokenfile=None, log=None):
-    """Determine the access token from the mounted secret or environment."""
+    """Determine the access token from the mounted configmap (nublado2),
+    secret (nublado1), or environment (either).  Prefer the mounted version
+    since it can be updated, while the environment variable stays at whatever
+    it was when the process was started."""
     tok = None
-    hdir = os.environ.get("HOME", None)
-    if hdir:
-        if not tokenfile:
-            # FIXME we should make this instance-dependent
-            tokfile = hdir + "/.access_token"
+    if tokenfile:
+        # If a path was specified, trust it.
+        token_path=Path(tokenfile)
+        return token_path.read_text()
+    else:
+        # Try the default token paths, nublado2 first, then nublado1
+        n2_tokenfile = "/opt/lsst/software/jupyterlab/environment/ACCESS_TOKEN"
+        token_path=Path(n2_tokenfile)
         try:
-            with open(tokfile, "r") as f:
-                tok = f.read().replace("\n", "")
-        except Exception as exc:
-            if not log:
-                log = make_logger()
-            log.warn("Could not read tokenfile '{}': {}".format(tokfile, exc))
+            tok=token_path.read_text()
+        except:
+            # Not there.  Let's try the nublado1 path.  Maybe this is an
+            #  instance we haven't upgraded yet.  Swallow the exception and
+            #  try the older mount.
+            pass
+        if not tok:
+            # We'll try nublado1: ${HOME}/.access_token
+            hdir = os.environ.get("HOME", None)
+            if hdir:
+                tokfile = hdir + "/.access_token"
+                token_path=Path(tokfile)
+            try:
+                tok=token_path.read_text()
+            except:
+                # OK, it's not mounted at all.  Fall back to the environment.
+                pass
     if not tok:
+        if not log:
+            log = make_logger()
+        log.warn("Could not read mounted tokenfile; trying environment.")
```

Comment:
This looks incorrect for the use case from one's laptop, which is what the original issue from GPDF was about. In that case, providing the token in an environment variable is probably the primary UI, so it shouldn't prompt a warning.